### ISSUE TITLE
Official Docker Image Build that Sugget GPU Support Utilizes CPU Only

### ISSUE THREAD

**Issue by BorisPolonsky** [tensorflow/tensorflow#40972]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 20.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): official Docker image (tensorflow/tensorflow:1.12.3-gpu-py3)
- TensorFlow version: 1.12.3
- Python version: 3.5.2
- Installed using virtualenv? pip? conda?:
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: (unchanged as is in the official docker image)
- GPU model and memory: GTX1050, 4042MiB

**Describe the problem**
Official release of docker image `tensorflow/tensorflow:1.12.3-gpu-py3` suggests gpu support in its tag, yet it does not utilize GPU. 

I have `nvidia-docker` (the native support version, with the `--gpus all` syntax) installed correctly. I've verified it thorugh additional test in `tensorflow/tensorflow:1.15.2-gpu-py3`, it worked fine, unlike `tensorflow/tensorflow:1.12.3-gpu-py3`.
**Provide the exact sequence of commands / steps that you executed before running into the problem**
- Run python3 in a container created from `tensorflow/tensorflow:1.12.3-gpu-py3`
`docker run -it --rm --gpus all tensorflow/tensorflow:1.12.3-gpu-py3 python3`
- Execute the following code
```
import tensorflow as tf
with tf.device("gpu:0"):
     c=tf.constant(0)

with tf.Session() as sess:
     sess.run(c)
```

**Any other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

After executing code in container created from `tensorflow/tensorflow:1.12.3-gpu-py3`, the error suggests that there is no avaliable GPU device:
```
InvalidArgumentError (see above for traceback): Cannot assign a device for operation Const: Operation was explicitly assigned to /device:GPU:0 but available devices are [ /job:localhost/replica:0/task:0/device:CPU:0, /job:localhost/replica:0/task:0/device:XLA_CPU:0 ]. Make sure the device specification refers to a valid device.
	 [[node Const (defined at <stdin>:2)  = Const[dtype=DT_INT32, value=Tensor<type: int32 shape: [] values: 0>, _device="/device:GPU:0"]()]]
```

The code above works fine in `tensorflow/tensorflow:1.15.2-gpu-py3`.

**Comment by amahendrakar** [comment]
@BorisPolonsky,
Is there a specific reason you are using TF v1.12?

TF 1.x is not actively supported and since this issue is resolved with TF v1.15 can we close this issue? Thanks!

**Comment by BorisPolonsky** [comment]
> @BorisPolonsky,
> Is there a specific reason you are using TF v1.12?
> 
> TF 1.x is not actively supported and since this issue is resolved with TF v1.15 can we close this issue? Thanks!

There's no such reason using this. I just found that one of the official image does not work as suggest in its tag. It's up to you to decide if it should be fixed (or documented).
